### PR TITLE
Feature/sql challenge support

### PR DESCRIPTION
**Pull request by rostikjoystick**
**YouTrack task**: [#APPS-2412](https://vyahhi.myjetbrains.com/youtrack/issue/APPS-2412)

**Description List**:
* Add SQL quiz support in new container

### CODE REVIEW

File: app/src/main/java/org/stepik/android/view/step_quiz_code/ui/delegate/CodeQuizInstructionDelegate.kt [hunk 1]
```diff
@@ -57,5 +57,10 @@ class CodeQuizInstructionDelegate(
 
     fun setCodeDetailsData(step: Step, lang: String?) {
         stepQuizCodeDetailsAdapter.items = codeStepQuizDetailsMapper.mapToCodeDetails(step, lang)
+        stepQuizCodeDetails.visibility = if (stepQuizCodeDetailsAdapter.items.isEmpty()) {
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/view/step_quiz_code/ui/delegate/CodeQuizInstructionDelegate.kt]: changeVisibility

File: app/src/main/java/org/stepic/droid/di/step/StepComponent.kt [hunk 2]
```diff
@@ -16,8 +14,4 @@ interface StepComponent {
     }
 
     fun commentsComponentBuilder(): CommentsComponent.Builder
```
Comment (eadm) [inline app/src/main/java/org/stepic/droid/di/step/StepComponent.kt]: StepComponent is not needed anymore

File: app/src/main/java/org/stepic/droid/ui/adapters/StepFragmentAdapter.kt [hunk 3]
```diff
@@ -39,19 +37,7 @@ class StepFragmentAdapter(
...
-            args.putParcelable(AppConstants.KEY_STEP_BUNDLE, stepWrapper)
-            args.putParcelable(AppConstants.KEY_LESSON_BUNDLE, lessonData.lesson)
-            args.putParcelable(AppConstants.KEY_UNIT_BUNDLE, lessonData.unit)
-            args.putParcelable(AppConstants.KEY_SECTION_BUNDLE, lessonData.section)
-            fragment.arguments = args
-            fragment
-        } else {
-            StepFragment.newInstance(stepWrapper, lessonData)
-        }
+        return StepFragment.newInstance(stepWrapper, lessonData)
```
Comment (eadm) [inline app/src/main/java/org/stepic/droid/ui/adapters/StepFragmentAdapter.kt]: expression body

File: app/src/main/java/org/stepic/droid/di/step/StepComponent.kt (hunk 2 above)
Comment (rostikjoystick) [reply app/src/main/java/org/stepic/droid/di/step/StepComponent.kt]: This is still needed in old comments implementation

**Pull request merged**